### DISCUSSION TITLE
SolidJS porting

### DISCUSSION THREAD

**Discussion by rtritto** [TexteaInc/json-viewer#327] [General]
Create a new project based on SolidJS.

### Core:
[react](https://www.npmjs.com/package/react) --> [solid-js](https://www.npmjs.com/package/solid-js)
[@mui/material](https://www.npmjs.com/package/@mui/material) --> [@suid/material](https://www.npmjs.com/package/@suid/material)

### State manager:
[jotai](https://www.npmjs.com/package/jotai) --> [solid-jotai](https://www.npmjs.com/package/solid-jotai)

Note: I can help 😉

**Comment by rtritto** [comment]
FYI @pionxzh @Himself65

**Comment by pionxzh** [comment]
I think we won't change to other frameworks :/ May I know what's the problem you want to fix with this proposal?

**Reply by rtritto** [reply]
Can be create a new repository for SolidJS using this project as source

**Reply by pionxzh** [reply]
It's ok if you want to port this project to solid.js. Btw, I think we cannot use solid components in React directly, right?

**Reply by rtritto** [reply]
> It's ok if you want to port this project to solid.js.

I asked to share my idea and if anyone it's interested.

Can TexteaInc be used?

> Btw, I think we cannot use solid components in React directly, right?

Correct and have no sense because there are valid alternatives (equal and already ported) for SolidJS.

**Reply by pionxzh** [reply]
> Can TexteaInc be used?

What do you mean? I mean, it's like you can have a `rtritto/json-viewer-solid`. And we can cite/link to each other in README. So trademark and Textea are not related here.

**Reply by pionxzh** [reply]
lol I didn't see that "new project" in your description. My bad.

**Reply by pionxzh** [reply]
I have tried jotai recently, and it feels great. I will take a look on your PR this week, but I'm not sure about the progress and status. I can also continue the work 😄

**Reply by rtritto** [reply]
> > Can TexteaInc be used?
> 
> What do you mean? I mean, it's like you can have a `rtritto/json-viewer-solid`. And we can cite/link to each other in README. So trademark and Textea are not related here.

We can use `TexteaInc/json-viewer-solid` to have both projects on same place.

Otherwise I will create `rtritto/json-viewer-solid` as you write.

**Reply by rtritto** [reply]
> I have tried jotai recently, and it feels great. I will take a look on your PR this week, but I'm not sure about the progress and status. I can also continue the work 😄 

Nice to hear, it's powerful 😃

Maybe the start of this new project should wait until the PR is merged.

**Reply by pionxzh** [reply]
I see. It's ok to just put it under `rtritto/`.

**Comment by pionxzh** [comment]
Sounds good and I believe this is totally doable.
And you may not even have to rely on Material UI. We only used `Box`, `DataBox` and `InputBox` in JSON viewer. However, I'm considering gradually phasing them out, perhaps in V4.
